### PR TITLE
Fix: Fix `auth/me` and `users/{:userId}` API

### PR DESCRIPTION
**Pull request by kanta1207**
## Overview

This PR is aiming to fix the PR, following the change in backend with the issue #255.
Only `auth/me` and `users/{:userId}` API client are fixed, and the one for generating invitation link still remains unresolved.
## Changes

- Fix API response type to use in `getCurrentUserId` method 
- Fix API response type to use in `getUserById` method
- Fix corresponding tests
- Apply the change to `ProfilePage`

## Assignee Checklist:

- [x] The naming convention of the PR title is correct (See the comment at the top of this template)
- [x] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] Double-check the "Assignee Checklist"
- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [ ] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/lib/api/user/server/index.ts
Comment (kanta1207) [inline src/lib/api/user/server/index.ts:1]: Just removed `ITemporallyGetUserByResponse`

**Review by ShoeheyOt — APPROVED** [review]
Thank you for quick work! LGTM :)
I will start working on fixing  invitation link hash after this branch is merged to develop in order to avoid conflict

**Comment by kanta1207** [comment]
@ShoeheyOt 
Thank you for the quick review!

File: src/components/pages/ProfilePage.tsx [hunk 1]
```diff
@@ -22,8 +22,8 @@ export const ProfilePage = async () => {
       const { userId } = getCurrentUserIdResult.unwrap();
       const getUserByIdResult = await getUserById(userId);
       if (getUserByIdResult.ok) {
-        const { name } = getUserByIdResult.unwrap();
-        return name;
+        const { username } = getUserByIdResult.unwrap();
+        return username;
```
Comment (nick-y-ito) [inline src/components/pages/ProfilePage.tsx:26]: @kanta1207 cc @ShoeheyOt 

Sorry for commenting after merging.

I think the key name defined in the backend is incorrect.

Our [Web API Document](https://nishiki-tech.github.io/nishiki-documents/web-api/index.html#tag/user/paths/~1users~1%7BuserId%7D/get) says the key is `name`, not `username`.

The term `username` sounds like a unique identifier, for example, "kanta1207" on your GitHub account. What we want here is the "name of the user." So, naming it `username` is a bit misleading.

Comment (nick-y-ito) [reply src/components/pages/ProfilePage.tsx:26]: @hitohata @taniguchiiqqq 
Could you please confirm this when you have a moment?

Comment (hitohata) [reply src/components/pages/ProfilePage.tsx:26]: @nick-y-ito 
I've created a new issue. Sorry for your inconvenience.
https://github.com/nishiki-tech/nishiki-backend/issues/315

Comment (kanta1207) [reply src/components/pages/ProfilePage.tsx:26]: @nick-y-ito Thank you for bringing it up!

I will quickly change it to `name` from `username` once it's resolved in backend, 
thank you @hitohata for the quick response!

Comment (nick-y-ito) [reply src/components/pages/ProfilePage.tsx:26]: @hitohata @kanta1207
Thank you for your prompt actions!

**Pull request merged**